### PR TITLE
docs: Update async race rules to make ponts more fair

### PR DESCRIPTION
**Pull request by lexarudak**
#### Title of Pull Request

Updating the requirements and evaluation of the epam/async race task

#### 🤔 This is a ...

- [ ] 🌟 New task
- [ ] 🌐 New module
- [x] ⚙️ Update to an existing task
- [ ] 🔧 Update to an existing module
- [ ] 🔗 Update or addition of external resources or links
- [ ] 🐛 Fix in a task or related content
- [ ] 🛠 Fix in a module or related content
- [ ] ✏️ Fixed a typo or grammatical error
- [ ] 🔗 Fixed a broken link
- [ ] ❓ Other (specify: **\*\*\*\***\_\_\_\_**\*\*\*\***)

#### Description

- **Brief Overview:**
-- For React, the use of a state manager has been made mandatory
-- The description of the page content has been made more detailed
-- The points for completed items have been redistributed. Points related to the implementation of the race have been made significantly more expensive
-- Removed items, the implementation of which when using a "framework" is either very simple or automatic. (For example, implementing SPA or "All HTML content must be dynamically generated using JavaScript")
- **Implementation Approach:**
  -- Change .md file)

### CODE REVIEW

File: epam/async-race.md [hunk 1]
```diff
@@ -119,82 +142,63 @@ Also, please deploy the UI to GitHub Pages, Netlify, Vercel, Cloudflare Pages, o
...
-## 📦 Bundling and Tooling (20 points)
-
-- [ ] **Use of Webpack or Similar (20 points):** Implement Webpack or another bundling tool to compile the project into a minimal set of files, ideally one HTML file, one JS file, and one CSS file. Ensure that the configuration enforces TypeScript strict type checking.
-
-## ✅ Code Quality and Standards (15 points)
-
-- [ ] **Eslint with Airbnb Style Guide (15 points):** Code must adhere to the Airbnb ESLint configuration to maintain code quality, as outlined in the [Airbnb style guide](https://www.npmjs.com/package/eslint-config-airbnb). Specific rules may be adjusted only with mentor approval, and there should be no ESLint errors or warnings.
-
-## 📏 Code Organization and Efficiency (15 points)
+## 🚗 Race (170 points)
```
Comment (lexarudak) [inline epam/async-race.md:178]: This is the most important and complex part of the task. Its assessment has been increased.

File: epam/async-race.md [hunk 2]
```diff
@@ -119,82 +142,63 @@ Also, please deploy the UI to GitHub Pages, Netlify, Vercel, Cloudflare Pages, o
...
+- [ ] **Button States (20 points):** Start engine button should be disabled in case car is already in driving mode. As well as stop engine button should be disabled when car is on it's initial place.
+- [ ] **Actions during the race (50 points):** Control over actions during a running race. Such as, deleting or editing a car, changing a page or view. Adding new cars. You can block buttons and stop the race. The main thing is to ensure predictable operation of the application.
 
 ## 🎨 Prettier and ESLint Configuration (10 points)
 
 - [ ] **Prettier Setup (5 points):** Prettier is correctly set up with two scripts in `package.json`: `format` for auto-formatting and `ci:format` for checking issues.
 - [ ] **ESLint Configuration (5 points):** ESLint is configured with the [Airbnb style guide](https://www.npmjs.com/package/eslint-config-airbnb). A `lint` script in `package.json` runs ESLint checks. Configuration files should reflect strict TypeScript settings as per `tsconfig.json`.
 
-## 🌟 Overall Code Quality (35 points)
+## 🌟 Overall Code Quality. (100 points) *Skip during self-check*
```
Comment (lexarudak) [inline epam/async-race.md:194]: The assessment has increased because more criteria have been added to this item.

File: epam/async-race.md [hunk 3]
```diff
@@ -119,82 +142,63 @@ Also, please deploy the UI to GitHub Pages, Netlify, Vercel, Cloudflare Pages, o
...
+- [ ] **Winners Table (15 points):** The table should include columns for the car's №, image, name, number of wins, and best time in seconds. If the same car wins more than once the number of wins should be incremented while best time should be saved only if it's better than the stored one.
 - [ ] **Sorting Functionality (10 points):** Allow users to sort the table by the number of wins and best time, in ascending or descending order.
 
-## 🏗️ Application Architecture (40 points)
-
-- [ ] **Modular Design (40 points):** The application should be clearly divided into logical modules or layers, such as API interaction, UI rendering, and state management. Consultation with a mentor on the architecture before implementation is advised.
-
-## 📜 Dynamic Content Generation (30 points)
-
-- [ ] **JavaScript-Generated HTML Content (30 points):** All HTML content must be dynamically generated using JavaScript, with the `<body>` tag containing only a single `<script>` tag.
```
Comment (lexarudak) [inline epam/async-race.md:171]: This is implied when using react.

File: epam/async-race.md [hunk 4]
```diff
@@ -119,82 +142,63 @@ Also, please deploy the UI to GitHub Pages, Netlify, Vercel, Cloudflare Pages, o
...
-
-- [ ] **SPA Implementation (25 points):** The application must be a Single Page Application (SPA) using either React v18+ or Angular v17+. All content must be generated using TypeScript with `strict` and `noImplicitAny` settings enabled in `tsconfig.json`, ensuring seamless user experience without page reloads during navigation.
-
-## 📦 Bundling and Tooling (20 points)
-
-- [ ] **Use of Webpack or Similar (20 points):** Implement Webpack or another bundling tool to compile the project into a minimal set of files, ideally one HTML file, one JS file, and one CSS file. Ensure that the configuration enforces TypeScript strict type checking.
-
-## ✅ Code Quality and Standards (15 points)
-
-- [ ] **Eslint with Airbnb Style Guide (15 points):** Code must adhere to the Airbnb ESLint configuration to maintain code quality, as outlined in the [Airbnb style guide](https://www.npmjs.com/package/eslint-config-airbnb). Specific rules may be adjusted only with mentor approval, and there should be no ESLint errors or warnings.
```
Comment (lexarudak) [inline epam/async-race.md:183]: This item was duplicated. there is the same one below in the file.

File: epam/async-race.md [hunk 5]
```diff
@@ -119,82 +142,63 @@ Also, please deploy the UI to GitHub Pages, Netlify, Vercel, Cloudflare Pages, o
...
-
-- [ ] **JavaScript-Generated HTML Content (30 points):** All HTML content must be dynamically generated using JavaScript, with the `<body>` tag containing only a single `<script>` tag.
-
-## 🌐 Single Page Application (25 points)
-
-- [ ] **SPA Implementation (25 points):** The application must be a Single Page Application (SPA) using either React v18+ or Angular v17+. All content must be generated using TypeScript with `strict` and `noImplicitAny` settings enabled in `tsconfig.json`, ensuring seamless user experience without page reloads during navigation.
-
-## 📦 Bundling and Tooling (20 points)
-
-- [ ] **Use of Webpack or Similar (20 points):** Implement Webpack or another bundling tool to compile the project into a minimal set of files, ideally one HTML file, one JS file, and one CSS file. Ensure that the configuration enforces TypeScript strict type checking.
```
Comment (lexarudak) [inline epam/async-race.md:179]: Vite or next collects the project out of the box. Plus, we have requirements for deploying the application. It doesn't seem like this needs to be evaluated separately.

File: epam/async-race.md [hunk 6]
```diff
@@ -119,82 +142,63 @@ Also, please deploy the UI to GitHub Pages, Netlify, Vercel, Cloudflare Pages, o
...
-
-- [ ] **Modular Design (40 points):** The application should be clearly divided into logical modules or layers, such as API interaction, UI rendering, and state management. Consultation with a mentor on the architecture before implementation is advised.
-
-## 📜 Dynamic Content Generation (30 points)
-
-- [ ] **JavaScript-Generated HTML Content (30 points):** All HTML content must be dynamically generated using JavaScript, with the `<body>` tag containing only a single `<script>` tag.
-
-## 🌐 Single Page Application (25 points)
-
-- [ ] **SPA Implementation (25 points):** The application must be a Single Page Application (SPA) using either React v18+ or Angular v17+. All content must be generated using TypeScript with `strict` and `noImplicitAny` settings enabled in `tsconfig.json`, ensuring seamless user experience without page reloads during navigation.
```
Comment (lexarudak) [inline epam/async-race.md:175]: It doesn't seem like this needs to be evaluated separately.

File: epam/async-race.md [hunk 7]
```diff
@@ -37,34 +37,57 @@ What are you waiting for? Let's start coding!
 
 ### Basic Structure
 
-- Two main views: "Garage" and "Winners", each with their name, page number, and a count of items in the database.
-- Persistent view state between switches, maintaining user input and pagination.
+- Two main views: "Garage" and "Winners.
+- Persistent state of user interface between switches, maintaining user input and pagination.
+- The race condition when switching between views can be handled to your liking (the race can be reset, continue, or stop). The main thing is to do it correctly.
```
[the remark this answers is not in the document]
Comment (dzmitry-varabei) [reply epam/async-race.md:42]: > It was important to separate the state of inputs and pagination from the state of the race during view switching.

Great, thanks! Let's just change the wording a little bit. 

### Basic Structure

- Two main views: "Garage" and "Winners."
- Persistent state of the user interface between switches, maintaining user input and pagination.
- The race condition when switching between views can be handled according to your preference (the race can be reset, continue, or stop). The main thing is to handle it correctly.

**Pull request merged**